### DISCUSSION TITLE
Keep original headers

### DISCUSSION THREAD

**Discussion by fm** [simple-login/app#1711] [Feature Requests]
Because SL creates a completely new email on forwarding, we lose all the header history of the original email. As an email engineer, I often need to look at headers for a variety of reasons a mere mortal normally wouldn't. Would be nice if we had a toggle to preserve the original headers for 14 days, similar to how the logs of senders/recipients is already kept. 

It would also be nice to see the SMTP response of the server when sending emails, to make sure it was delivered or if its delayed due to greylisting. 

ImprovMX handles it beautifully, as seen below:

<img width="958" alt="Screenshot 2023-04-29 at 3 48 17 PM" src="https://user-images.githubusercontent.com/3654256/235306285-51427b05-f6cb-41af-a56d-2a2722264992.png">

**Comment by nrogovik** [comment]
This would be nice. We're encouraged to use these aliases for all our online accounts. And half an email header is pretty much useless if we need to check them for whatever reason. Got my vote.

**Comment by Edontin** [comment]
This would definitely be useful to investigate phishing attempts.
I'd also find it helpful if some headers could be copied from the original email into an `X-SimpleLogin-Original-<Name>` header, like the envelope headers that exist today. Some of my Sieve rules are reliant on the `Precedence` or `List-Id` headers for classifying emails.

**Comment by locallylex** [comment]
Also bumping this. Half the functionality of sieve filters is unavailable when using aliases because headers like `List-Unsubscribe`, `X-Campaign`, and `Precedence` are not available after forwarding.

**Comment by felonbyfiat** [comment]
This really needs to be added as a feature. If anything; retain the original headers for at least 14 days in the simple login dashboard.

I've seen several emails where I need to check the headers for origin and other details but I am unable to do so.

**Comment by laurens94** [comment]
When reporting spam/phishing emails, providers often ask for original email headers. It would be great if we were able to provide those when using simplelogin.
